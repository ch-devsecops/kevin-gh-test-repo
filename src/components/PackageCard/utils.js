/* eslint-disable no-param-reassign */
const addUnOrderedListPadding = (component, type) => {
  if (component?.componentName === 'UnOrderedList') {
    if (!component.params) component.params = {};

    component.params.px = type !== 'nested' ? ['20px', 'zero'] : 'zero';
    component.params.pb = 'zero';
  }
};

const addFreeFormTextMargin = component => {
  if (component?.componentName === 'FreeFormText') {
    if (!component.params) component.params = {};

    component.params.mb = 'zero';
  }
};

/**
 * PackageCard utility helper that handles spacing of
 * UnorderedLists and FreeFormText in a components array
 */
export const handleComponentSpacing = components =>
  components.map(component => {
    if (component.props && component.props.type === 'text/sitecore') {
      return component;
    }

    const rendering = component?.props?.rendering;

    addUnOrderedListPadding(rendering);
    addFreeFormTextMargin(rendering);

    // handles placeholder components
    const placeholderNames = rendering?.placeholders ? Object.keys(rendering.placeholders) : [];

    if (placeholderNames.length) {
      placeholderNames.forEach(name => {
        rendering.placeholders[name].forEach(c => {
          addUnOrderedListPadding(c, 'nested');
          addFreeFormTextMargin(c);
        });
      });
    }

    return component;
  });

export default { handleComponentSpacing };
