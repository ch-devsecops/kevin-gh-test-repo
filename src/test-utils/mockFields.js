/* eslint-disable max-len */
/* eslint-disable max-classes-per-file */

// FIXME: jsdoc popup not working on test files
/** Class representing a mock data. */
class MockData {
  /**
   * Add a field.
   * @param {string} key - key of the field obj: i.e. 'title'
   * @param {Object} value - value of the field obj: i.e. { value: 'some text' }.
   */
  add(key, value) {
    this[key] = value;
  }
}

/**
 * Class representing mock fields object.
 * @extends MockData
 *
 * available method: add()
 *
 * default fields: [anchorId, title, bodyText, image, ctaLink, ctaType, ctaIcon, gtmCategory, gtmTitle, contentTitle, subtitle]
 *
 * note: we can add more commonly used fields in the class declaration if needed
 */
class MockFields extends MockData {
  constructor() {
    super();

    this.title = { value: 'Title' };
    this.bodyText = { value: 'BodyText' };
    this.image = {
      value: { src: 'mock.png', alt: 'mock alt' },
    };
    this.ctaLink = {
      value: {
        href: 'www.source.ca',
        title: 'cta aria label',
        text: 'CTA Text',
      },
    };
    this.ctaType = {
      value: 'Tertiary',
    };
    this.ctaIcon = {
      fields: {
        value: {
          value: 'arrowRight',
        },
      },
    };
    this.gtmCategory = {
      fields: {
        value: {
          value: 'Others',
        },
      },
    };
    this.gtmTitle = {
      value: 'gtm title',
    };
    this.contentTitle = {
      value: 'Content Title',
    };
    this.subtitle = {
      value: 'Content Subtitle',
    };
    this.seoH1 = {
      value: '',
    };
    this.anchorId = {
      value: 'anchor-id',
    };
    this.ctaLabel = {
      value: 'CTA Label',
    };
  }
}

/**
 * Class representing a mock rendering object
 * @extends MockData
 *
 * available method: add()
 */
class MockRendering extends MockData {
  /**
   * Create a mock rendering object
   * @param {string} componentName - adds componentName
   */
  constructor(componentName) {
    super();

    this.componentName = componentName;
  }
}

export { MockFields, MockRendering };
