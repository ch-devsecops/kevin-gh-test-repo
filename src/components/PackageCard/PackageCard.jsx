import React from 'react';
import PropTypes from 'prop-types';

import { Box, Markdown, Tooltip, Icon, useThemeContext, Media } from '@honda-canada/design-system-react';
import { ACURA_THEME_NAME } from '../../utils/constants';
import themeStyles from './PackageCard.styles';

const Container = themeStyles.apply(Box, 'Container');
const FeaturedText = themeStyles.apply(Box, 'FeaturedText');
const Title = themeStyles.apply(Box, 'Title');
const PricingContainer = themeStyles.apply(Box, 'PricingContainer');
const HighlightText = themeStyles.apply(Box, 'HighlightText');
const BannerText = themeStyles.apply(Box, 'BannerText');
const SpecialOfferContainer = themeStyles.apply(Box, 'SpecialOfferContainer');

const PackageCard = ({
  featuredText,
  title,
  bannerText,
  highlightText,
  subtitle1,
  subtitle2,
  amount,
  tooltip = {},
  specialOfferLabel,
  specialOfferPrice,
  anchorId,
  linked = false,
  gtmTags = {},
  layoutContainer,
  children,
}) => {
  const { name } = useThemeContext();
  const hasPricingBanner = highlightText || specialOfferPrice;
  const copyColor = hasPricingBanner && 'white';
  const copySize = name === ACURA_THEME_NAME ? 'small' : 'regular';
  const copyFamily = name === ACURA_THEME_NAME ? 'default' : 'bold';
  const tooltipStyling = hasPricingBanner ? 'dark' : 'light';
  const tooltipTextColor = tooltipStyling === 'light' && 'white';
  const inBiggerColumn =
    layoutContainer === 'SingleColumnLayout' ||
    layoutContainer === 'TwoColumnLayout' ||
    layoutContainer === 'LayoutContainer'; // for local dev

  return (
    <Container
      id={anchorId}
      data-gtm-component-type={gtmTags.type}
      data-gtm-category={gtmTags.category}
      inBiggerColumn={inBiggerColumn}
    >
      {featuredText && (
        <FeaturedText>
          <Markdown color="white">{featuredText}</Markdown>
        </FeaturedText>
      )}

      <Title linked={linked}>{title}</Title>

      {linked && (
        <BannerText hasContent={bannerText}>
          <Box maxWidth="514px">
            {bannerText && (
              <Markdown fontWeight="bold" fontFamily="heading">
                {bannerText}
              </Markdown>
            )}
          </Box>
        </BannerText>
      )}

      <PricingContainer hasBanner={hasPricingBanner} linked={linked}>
        {highlightText && (
          <HighlightText>
            <Markdown color={copyColor}>{highlightText}</Markdown>
          </HighlightText>
        )}

        {specialOfferPrice && (
          <Media lessThan="desktop">
            <Markdown color={copyColor} mb="xxs">
              {specialOfferLabel}
            </Markdown>
          </Media>
        )}

        <Markdown color={copyColor} size={copySize} fontFamily={copyFamily} mb={linked ? ['xxs', 'm'] : undefined}>
          {subtitle1}
        </Markdown>

        <Box position="relative" display="flex">
          {specialOfferPrice && (
            <SpecialOfferContainer>
              <Box
                // handles spacing from center of special offers container
                display="flex"
                justifyContent="flex-end"
                pr={['60px', '90px']}
              >
                <Media greaterThan="smallDesktop">
                  <Markdown pr="default" color={copyColor}>
                    {specialOfferLabel}
                  </Markdown>
                </Media>

                <Box>
                  <Markdown color={copyColor} size="pricing" textDecoration="line-through">
                    {specialOfferPrice}
                  </Markdown>
                </Box>
              </Box>
            </SpecialOfferContainer>
          )}

          <Box width="100%" display="flex" justifyContent="center">
            <Markdown color={copyColor} fontFamily="heading" size="pricingLarge" mb={linked ? ['xxs', 'm'] : undefined}>
              {amount}
            </Markdown>

            {tooltip?.content && (
              <Tooltip
                content={
                  <Markdown size="xSmall" color={tooltipTextColor}>
                    {tooltip.content}
                  </Markdown>
                }
                ariaLabel={tooltip.ariaLabel}
                closeAriaLabel={tooltip.closeAriaLabel}
                styling={tooltipStyling}
                ml="xxs"
                my={['xs', 's']}
                forceHorizontalPosition="left"
              >
                {({ active }) => (
                  <Box
                    width="16px" // iOS mobile / tablet fix
                    height="16px"
                  >
                    <Icon name="information" filled={active} inverted={tooltipStyling === 'dark'} />
                  </Box>
                )}
              </Tooltip>
            )}
          </Box>
        </Box>

        <Markdown color={copyColor} size={copySize} fontFamily={copyFamily}>
          {subtitle2}
        </Markdown>
      </PricingContainer>

      {children}
    </Container>
  );
};

PackageCard.propTypes = {
  /**
   * Top banner text
   */
  featuredText: PropTypes.string,
  /**
   * Text that Handles edge to edge banner
   */
  highlightText: PropTypes.string,
  bannerText: PropTypes.string,
  title: PropTypes.oneOfType([PropTypes.string, PropTypes.element]),
  subtitle1: PropTypes.string,
  subtitle2: PropTypes.string,
  amount: PropTypes.string,
  tooltip: PropTypes.string,
  specialOfferLabel: PropTypes.string,
  specialOfferPrice: PropTypes.string,
  anchorId: PropTypes.string,
  /**
   * If true, keeps the banner text container even if banner text is not provided
   */
  linked: PropTypes.bool,
  gtmTags: PropTypes.shape({
    type: PropTypes.string,
    category: PropTypes.string,
  }),
  layoutContainer: PropTypes.string,
  children: PropTypes.element,
};

export default PackageCard;
