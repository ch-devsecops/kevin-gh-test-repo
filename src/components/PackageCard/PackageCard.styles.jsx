import css from '@styled-system/css';
import { ThemeStyles } from '@honda-canada/design-system-react';
import { MC_THEME_NAME } from '../../utils/constants';

const themeStyles = new ThemeStyles();

themeStyles.addBaseStyles('Container')(
  ({ inBiggerColumn }) =>
    css({
      border: '1px solid',
      borderColor: 'grey.2',
      backgroundColor: 'white',
      pt: inBiggerColumn ? ['xl', 'xxl'] : 'xl',
    }),
  {
    width: '100%',
    height: '100%',
    display: 'flex',
    flexDirection: 'column',
    alignItems: 'center',
    position: 'relative',
  },
);

themeStyles.addBaseStyles('FeaturedText')(({ theme }) =>
  css({
    width: '100%',
    height: ['default', '36px'],
    textAlign: 'center',
    backgroundColor: theme.name === MC_THEME_NAME ? 'red' : 'blue',
    color: 'white',
    minWidth: '280px',
    maxWidth: ['100%', '280px'],
    py: ['xxs', 'xs'],

    // center absolute position
    position: 'absolute',
    left: 0,
    right: 0,
    top: 0,
    ml: 'auto',
    mr: 'auto',
  }),
);

themeStyles.addBaseStyles('Title')(({ linked }) =>
  css({
    width: '100%',
    minHeight: ['initial', linked ? '56px' : 'initial'],
    textAlign: 'center',
    mb: ['15px', 'default'],
    px: 's',
  }),
);

themeStyles.addBaseStyles('PricingContainer')(
  ({ theme, hasBanner }) => {
    const bgColor = theme.name === 'mc' ? 'red' : 'darkBlue';

    return css({
      backgroundColor: hasBanner ? bgColor : 'white',
      py: hasBanner ? ['xs', 'm'] : 0,
      mb: hasBanner ? 'default' : ['default', 'l'],
    });
  },
  ({ linked, hasBanner }) =>
    linked &&
    css({
      minHeight: [hasBanner ? '131px' : '144px', '222px', '186px'], // iOS and macOs fix
      maxHeight: ['131px', '222px', '186px'],
      height: '100%',
      py: [hasBanner ? 'xs' : 'default', 'default'],
      display: 'flex',
      flexDirection: 'column',
      alignItems: 'center',
      justifyContent: 'center',
      mb: '0 !important',

      // in IE: fix overflowing text
      '@media screen and (-ms-high-contrast: active), (-ms-high-contrast: none)': {
        minHeight: '186px',
      },
    }),
  {
    textAlign: 'center',
    width: '100%',
  },
);

themeStyles.addBaseStyles('SpecialOfferContainer')(
  css({
    alignContent: 'flex-start',
    position: 'absolute',
    width: 1 / 2,
    py: ['xxs', 'xs'],

    // in IE
    '@media screen and (-ms-high-contrast: active), (-ms-high-contrast: none)': {
      left: '-124px',
    },
  }),
);

themeStyles.addBaseStyles('HighlightText')(
  css({
    mb: ['xs', 'm'],
  }),
);

themeStyles.addBaseStyles('BannerText')(({ hasContent }) =>
  css({
    px: ['m', 'default'],
    mb: ['zero', hasContent && 'default', 'default'],
    width: '100%',
    textAlign: 'center',
    display: 'flex',
    alignItems: 'center',
    justifyContent: 'center',
    height: hasContent ? '76px' : [0, 0, '76px'],
    minHeight: hasContent ? '76px' : [0, 0, '76px'],
    bg: hasContent ? 'grey.3' : 'white',
  }),
);

export default themeStyles;
