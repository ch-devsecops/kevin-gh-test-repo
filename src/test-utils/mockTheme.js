/**
 * Mock styled-components theme, if you want to test how your component
 * implements theme styles. Import as-is, or spread in any
 * additional values you need.
 *
 * Example usage:

  it('should apply the bgColor param', () => {
    const { colors } = MockTheme;
    const { container } = render(
      <MyComponent
        fields={mockFields}
        params={{ bgColor: 'Gray' }}
      />,
    );
    const wrapper = container.querySelector('section');

    expect(wrapper).toHaveStyleRule('background-color', colors.grey[5]);
  });
 */

import { HONDA_THEME_NAME } from '../utils/constants';

const mockTheme = {
  letterSpacings: {
    sm: '0.5px',
    md: '1px',
  },
  lineHeights: {
    sm: '22px',
    md: '26px',
  },
  colors: {
    white: '#fff',
    transparent: 'transparent',
    grey: ['#707070', '#C7C7C7', '#D5D5D5', '#EAEAEA', '#F8F8F8', '#F5F5F5'],
    black: '#2C2C2C',
    red: '#DA2C2E',
    redShades: ['rgba(200, 16, 46, 0.1)'],
    aodaRed: '#D92629',
    lightRed: '#EC9597',
    darkRed: '#A21C1D',
    blue: '#334C96',
    darkBlue: '#002D62',
    lightBlue: '#00447826',
    yellow: ['#F9CA52', '#EDAB00', '#8D763F'],
    fadedBlue: '#ADB7D5',
    primary: '#DA2C2E',
    primaryAoda: '#D92629',
    primaryDark: '#A21C1D',
    primaryLight: '#EC9597',
    secondary: '#2C2C2C',
    secondaryLight: '#C7C7C7',
    tertiary: '#002D62',
    tertiaryLight: '#334C96',
    tertiaryDisabled: '#ADB7D5',
    danger: '#DA2C2E',
    warning: '#2C2C2C',
    warningLight: '#EAEAEA',
    success: '#334C96',
    successLight: '#d9e0e7',
    disabled: '#C7C7C7',
    typographyDefault: '#2C2C2C',
  },
  space: [0, 4, 8, 12, 16, 24, 32, 48, 64, 96, 128, 144, 192, 256],
  borders: [0, '1px solid', '2px solid', '3px solid'],
  breakpoints: ['768px', '992px', '1200px'],
  fontWeights: {
    regular: 'normal',
    bold: 700,
  },
  fontSizes: {
    sm: '12px',
  },
  shadows: {
    smooth: {
      md: '0.1px 0.3px 0.3px rgba(0, 0, 0, 0.02),0.3px 0.7px 0.9px rgba(0, 0, 0, 0.028),0.7px 1.4px 1.8px rgba(0, 0, 0, 0.035),1.5px 2.9px 3.7px rgba(0, 0, 0, 0.047),4px 8px 10px rgba(0, 0, 0, 0.07)',
    },
    box: {
      subNav: '0px 2px 4px rgba(0, 0, 0, 0.1)',
    },
  },
  zIndices: {
    accordionItemTitle: 200,
    modal: 13001,
    sectionNav: 13000,
  },
  transitionDuration: {
    t1: '1s',
  },
  transitionTimingFunction: {
    default: 'ease-in-out',
    out: 'ease-out',
    in: 'ease-in',
  },
  name: HONDA_THEME_NAME,
  fonts: {
    default: 'Helvetica',
    bold: 'Cooper Black',
    heading: 'Roboto',
  },
  header: {
    desktop: {},
    mobile: {},
  },
  icons: {},
  logIconWarnings: false,
};

export default mockTheme;
