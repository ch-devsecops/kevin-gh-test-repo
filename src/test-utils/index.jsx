import React from 'react';
import { render } from '@testing-library/react';
import { ThemeProvider } from 'styled-components';
import { SitecoreContext } from '@sitecore-jss/sitecore-jss-react';
import { BrowserRouter } from 'react-router-dom';
import { OptionalVideoModalContextProvider } from '@honda-canada/design-system-react';
import UserLocationProvider from '@honda-canada/user-location';
import { configureAxe } from 'jest-axe';
import ModelExplorationContextProvider from '../components/ModelExplorationContext';
import ModelFiltersContextProvider from '../components/ModelFiltersContext';
import { MockFields, MockRendering } from './mockFields';
import MockTheme from './mockTheme';
import 'intersection-observer';
import { ACURA_THEME_NAME } from '../utils/constants';

const generateJssWrapper = context =>
  // eslint-disable-next-line react/display-name
  function ({ children }) {
    return (
      <SitecoreContext
        context={{
          language: 'en',
          site: { name: ACURA_THEME_NAME },
          ...context?.sitecoreContext,
        }}
        componentFactory={{}}
      >
        <BrowserRouter>
          <UserLocationProvider>
            <ThemeProvider theme={context?.theme || MockTheme}>
              <OptionalVideoModalContextProvider>
                <ModelExplorationContextProvider>
                  <ModelFiltersContextProvider>{children}</ModelFiltersContextProvider>
                </ModelExplorationContextProvider>
              </OptionalVideoModalContextProvider>
            </ThemeProvider>
          </UserLocationProvider>
        </BrowserRouter>
      </SitecoreContext>
    );
  };
/**
 * A custom implementation of the testing-library `render` function that
 * wraps tests in an typical JSS app's context providers.
 * Note that you can include a `context` object to set or override values
 * for SitecoreContext and ThemeContext.
 * @param {ReactElement} component - the component you want testing-library to render
 * @param {object} [options] - see https://testing-library.com/docs/react-testing-library/api/#render-options
 * @param {string} [options.context] - default values for SitecoreContext, ThemeProvider
 * @param {string} [options.context.sitecoreContext] - value for SitecoreContext
 * @param {string} [options.context.theme] - value for ThemeProvider
 * @example
 * const { getByText } = jssRender(
 *  <MyComponent />, { context: { sitecoreContext: { language: 'fr' } } }
 * );
 * @returns {function(): void} custom testing-library render function
 */
const jssRender = (component, options = {}) => {
  const { context } = options;
  // eslint-disable-next-line no-param-reassign
  delete options.context;

  return render(component, {
    wrapper: generateJssWrapper(context),
    ...options,
  });
};

const generateWrapper = context =>
  // eslint-disable-next-line react/display-name
  function ({ children }) {
    return (
      <BrowserRouter>
        <ThemeProvider theme={context?.theme || MockTheme}>
          <OptionalVideoModalContextProvider>{children}</OptionalVideoModalContextProvider>
        </ThemeProvider>
      </BrowserRouter>
    );
  };

/**
 * A custom implementation of the testing-library `render` function that
 * wraps tests in typical design-system context providers.
 * Note that you can include a `theme` object to set or override values
 * for ThemeContext.
 * @param {ReactElement} component - the component you want testing-library to render
 * @param {object} [options] - see https://testing-library.com/docs/react-testing-library/api/#render-options
 * @param {string} [options.theme] - default value for ThemeProvider
 * @example
 * const { getByText } = render(
 *  <MyComponent />, { theme: { space: { wrapperGap: '15px' } } }
 * );
 * @returns {function(): void} custom testing-library render function
 */
const customRender = (component, options = {}) => {
  const { theme } = options;
  // eslint-disable-next-line no-param-reassign
  delete options.theme;

  return render(component, {
    wrapper: generateWrapper(theme),
    ...options,
  });
};

// disable accessibility checks for errors that are usually trigged by content mistakes
const axe = configureAxe({
  rules: {
    'duplicate-id': { enabled: false },
    'heading-order': { enabled: false },
  },
});

export * from 'jest-axe';
export { axe };
export * from '@testing-library/react';
export { customRender as render, jssRender };
export { MockFields, MockRendering, MockTheme };
