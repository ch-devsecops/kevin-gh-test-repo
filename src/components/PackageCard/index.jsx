import React, { useContext } from 'react';
import { Box, H5 } from '@honda-canada/design-system-react';
import { Placeholder } from '@sitecore-jss/sitecore-jss-react';
import { compiler } from 'markdown-to-jsx';
import { getTitleComponent, stripMarkdownHeading } from '../../utils/markdown';
import { wrapJSSFields } from '../../utils/wrapJSSFields';
import { mapGTMCategory } from '../../utils/sitecoreFields';
import { handleComponentSpacing } from './utils';
import { LayoutContext } from '../LayoutContext';
import PackageCard from './PackageCard';

const PackageCardJSS = ({ fields, rendering }) => {
  const { layoutName } = useContext(LayoutContext);

  if (!fields) return null;

  const { featuredText, title, highlightText, subtitle1, subtitle2, amount, anchorId, gtmCategory } =
    wrapJSSFields(fields);

  const contentPlaceholder = rendering?.placeholders || {};
  const contentName = Object.keys(contentPlaceholder)[0];

  const gtmTags = {
    type: rendering?.componentName,
    category: mapGTMCategory(gtmCategory),
  };

  const TitleComponent = getTitleComponent(title?.value, H5);
  const TitleContent = <TitleComponent>{compiler(stripMarkdownHeading(title?.value))}</TitleComponent>;

  return (
    <PackageCard
      featuredText={featuredText?.value}
      title={TitleContent}
      highlightText={highlightText?.value}
      subtitle1={subtitle1?.value}
      subtitle2={subtitle2?.value}
      amount={amount?.value}
      anchorId={anchorId?.value}
      gtmTags={gtmTags}
      layoutContainer={layoutName}
    >
      <Placeholder
        name={contentName}
        rendering={rendering}
        render={(components, placeholderData) => {
          // handles UnOrderedList and FreeFormText spacing
          const componentsWithSpacing = handleComponentSpacing(components);

          // handles content container horizontal padding
          const allowedChildLayout = [
            'ThreeColumnLayout',
            'TwoColumnLayout',
            'LayoutContainers', // checks for layout containers locally
          ];
          const hasChildLayoutContainer = placeholderData.some(component =>
            allowedChildLayout.includes(component?.componentName),
          );

          // handles content table component bottom padding
          const hasChildTableComponent = placeholderData.some(component => component?.componentName === 'Table');

          const containerStyle = {};

          if (hasChildLayoutContainer) {
            containerStyle.px = ['0px', 'default', '106px'];
          }
          if (hasChildTableComponent) {
            containerStyle.mb = 'zero';
          }

          return (
            <Box
              width="100%"
              display="flex"
              flexDirection="column"
              justifyContent="center"
              alignItems="center"
              mb={['l', 'xxl']}
              {...containerStyle}
            >
              {componentsWithSpacing}
            </Box>
          );
        }}
      />
    </PackageCard>
  );
};

export default PackageCardJSS;
