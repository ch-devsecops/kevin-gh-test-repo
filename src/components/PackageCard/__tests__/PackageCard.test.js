import React from 'react';
import { axe, cleanup, render, MockRendering, MockFields } from '../../../test-utils';
import PackageCard from '../PackageCard';

afterEach(cleanup);

const mockFields = new MockFields();
const mockRendering = new MockRendering('PackageCard');

describe('<PackageCard>', () => {
  it('should render correctly with all props', async () => {
    const { container } = render(
      <PackageCard fields={mockFields} rendering={mockRendering} title="My title" bodyText="My body text" />,
    );

    expect(await axe(container)).toHaveNoViolations();
    expect(container).toMatchSnapshot();
  });
});
