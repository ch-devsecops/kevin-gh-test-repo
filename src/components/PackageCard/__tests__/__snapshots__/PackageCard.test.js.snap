// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PackageCard> should render correctly with all props 1`] = `
<div>
  <div
    class="Box-sc-1bsf784-0 sc-1k44dfv-0 dNlNKC"
  >
    <div
      class="Box-sc-1bsf784-0 sc-1k44dfv-0 hhevUI"
    >
      My title
    </div>
    <div
      class="Box-sc-1bsf784-0 sc-1k44dfv-0 kTIOGw"
    >
      <div
        class="sc-1k44dfv-0 diivzi"
        font-family="bold"
      >
        

        <p>
          undefined
        </p>
      </div>
      <div
        class="Box-sc-1bsf784-0 toprL"
        display="flex"
      >
        <div
          class="Box-sc-1bsf784-0 NHKAQ"
          display="flex"
          width="100%"
        >
          <div
            class="sc-1k44dfv-0 iTQBMw"
            font-family="heading"
          >
            

            <p>
              undefined
            </p>
          </div>
        </div>
      </div>
      <div
        class="sc-1k44dfv-0 diivzi"
        font-family="bold"
      >
        

        <p>
          undefined
        </p>
      </div>
    </div>
  </div>
</div>
`;
